Add V2_importer to collect advisories from EUVD

Samk1710 opened the pull request:
- Fixes: [https://github.com/aboutcode-org/vulnerablecode/issues/1873](https://github.com/aboutcode-org/vulnerablecode/issues/1873)


### Data Source

* **API**: `https://euvdservices.enisa.europa.eu/api/search`
* **Format**: JSON with pagination

Dependent on : 
- https://github.com/aboutcode-org/aboutcode-mirror-euvd/pull/1

Logs:

```text
INFO 2025-11-25 23:53:04.724015 UTC Progress: 100% (452459/452459)
INFO 2025-11-25 23:53:04.735236 UTC Successfully collected 452,441 advisories
INFO 2025-11-25 23:53:04.735444 UTC Step [collect_and_store_advisories] completed in 23064 seconds (6.4 hours)
```

ziadhany commented on vulnerabilities/tests/pipelines/v2_importers/test_euvd_importer_v2.py:
I think it would be easier if you test using `util_tests.check_results_against_json(result, expected_file)` and with an expected file.

ziadhany commented on vulnerabilities/pipelines/v2_importers/euvd_importer.py:
```diff
@@ -0,0 +1,218 @@
+
+        logger.info(f"Fetching data from EUVD API: {self.url}")
+
+        while True:
```
We should avoid loops without a condition.

ziadhany reviewed (changes requested):
@Samk1710 Great start! Just a few small tweaks

Samk1710 commented:
Hey @ziadhany ,

I’ve pushed the requested updates. Summary of changes:

* Replaced broad exception handling with specific parsing-related exceptions.
* Updated the test to use `util_tests.check_results_against_json` along with expected JSON fixtures.
* Removed unbounded loops by using the API’s total count field to determine total number of pages.
* Simplified retry handling(Retains a fixed 3-second delay between retries to prevent rapid page skipping in case of network interruptions).

Let me know if you’d like any additional modifications. Thanks again for the feedback and guidance!

pombredanne commented on vulnerabilities/tests/pipelines/v2_importers/test_euvd_importer_v2.py:
```diff
@@ -0,0 +1,78 @@
+        expected_file = TEST_DATA / "euvd-expected.json"
+        util_tests.check_results_against_json(advisories, expected_file)
+
+    def test_get_scoring_system(self):
```
Please create one test function for each

Samk1710 commented:
The product field is a little messy. We might need to drop some(less than 1 %).

For eg:
- https://euvd.enisa.europa.eu/vulnerability/EUVD-2023-12722
- https://euvd.enisa.europa.eu/vulnerability/EUVD-2024-16938
- https://euvd.enisa.europa.eu/vulnerability/EUVD-2023-39960
- https://euvd.enisa.europa.eu/vulnerability/EUVD-2017-8416
